development/tools/pkg/vmscollector/wrappers.go

```diff
@@ -0,0 +1,47 @@
+package vmscollector
+
+import (
+	"context"
+
+	compute "google.golang.org/api/compute/v1"
+)
+
+// InstancesAPIWrapper abstracts GCP InstancesService API
+type InstancesAPIWrapper struct {
+	Context context.Context
+	Service *compute.InstancesService
+}
+
+// ListInstances delegates to InstancesService.ListInstances(project) function
+func (iaw *InstancesAPIWrapper) ListInstances(project string) ([]*compute.Instance, error) {
```

Comment:
Something we might want to think about is, if we want to filter something from the instances at API call level. `InstanceList` and `InstanceAggregatedList` support the filter function, so we might filter our instances of no interest to us (wrong state, no labels, etc)

Reply:
Good point for future improvement. For now let's give it a try and gather logs from production environment. Once we're sure we understand the environment well enough, such an improvement can be made.